odoo/api.py

```diff
@@ -631,25 +631,77 @@ def __call__(self, cr=None, user=None, context=None, su=None):
         su = (user is None and self.su) if su is None else su
         return Environment(cr, uid, context, su, self.uid_origin)
 
-    def ref(self, xml_id, raise_if_not_found=True):
-        """ Return the record corresponding to the given ``xml_id``.
+    def ref(self, *xml_ids: str, raise_if_not_found: bool = True) -> BaseModel | None:
+        """ Return the records corresponding to the given ``xml_ids``.
+
+        :param str xml_ids: record xml_ids, under the format ``<module>.<record_id>``
+        :param bool raise_if_not_found: whether the method should raise if any ``xml_id`` isn't found.
+        :returns: Found records or None
+        :raise ValueError: if any record wasn't and ``raise_if_not_found`` is True,
+            or if the user mixed ``xml_ids`` that belong to different models.
+
+        .. note::
+            If ``raise_if_not_found`` is ``False`` and there are multiple
+            ``xml_ids`` as input, then it won't be possible to determine which
+            specific records weren't found by only looking at the recordset.
+
+        .. note::
+            | To avoid breaking the API, now that ``raise_if_not_found`` is a keyword-only argument,
+              we check the last ``xml_ids``:
+            | if it's a ``bool`` then it's used as ``raise_if_not_found``.
+
+            For example::
+
+                self.env.ref('base.us', 'base.x', False)
+                res.country(233)
+
+                self.env.ref('base.us', 'base.x', raise_if_not_found=False)
+                res.country(233)
+
+                self.env.ref('base.us', 'base.x')
+                ValueError: External ID not found in the system: base.x
+
+                self.env.ref('base.us', 'base.x', True)
+                ValueError: External ID not found in the system: base.x
+
+                self.env.ref('base.us', 'base.x', raise_if_not_found=True)
+                ValueError: External ID not found in the system: base.x
 
-        :param str xml_id: record xml_id, under the format ``<module.id>``
-        :param bool raise_if_not_found: whether the method should raise if record is not found
-        :returns: Found record or None
-        :raise ValueError: if record wasn't found and ``raise_if_not_found`` is True
         """
-        res_model, res_id = self['ir.model.data']._xmlid_to_res_model_res_id(
-            xml_id, raise_if_not_found=raise_if_not_found
-        )
 
-        if res_model and res_id:
-            record = self[res_model].browse(res_id)
-            if record.exists():
-                return record
-            if raise_if_not_found:
-                raise ValueError('No record found for unique ID %s. It may have been deleted.' % (xml_id))
-        return None
+        # Avoid breaking the API when raise_if_not_found was given as a pure boolean non-keyword arg
+        if len(xml_ids) > 1 and isinstance(xml_ids[-1], bool):
+            raise_if_not_found = xml_ids[-1]
```

Comment:
If you do this, add a deprecation warning.

Reply:
Why am I such a people pleaser...
![immagine](https://github.com/user-attachments/assets/7122b1da-97cd-4893-a32a-b95f067a7dde)
I think I'm gonna add it.

Reply:
Should I add a DeprecationWarning, what and how should I write that on the documentation? Do we have an example?

Reply:
Looks good.

Reply:
The point of it was to allow both syntax (not have a keyword only param) and be fully backward compatible (hence my suggestion to not have a deprecation warning) :shrug:

Reply:
Keeping multiple APIs is bad design, it would mean we would leave for a long time the `if` to check if it is the new signature or the old one. The goal of the deprecation is exactly to have a period during which we are backward-compatible while the API changes.